app/views/songs/index.html.erb

```diff
@@ -1,29 +1,47 @@
-<h1>Songs</h1>
-<table class="songs-table">
-  <thead>
-    <tr>
-      <th>Name</th>
-      <th>Artist</th>
-      <th>Tempo</th>
-      <th>Key</th>
-    </tr>
-  </thead>
-</table>
 
+<%= stylesheet_link_tag 'songs_manifest' %>
+
+<div class="index-page">
+  <div class="container">
+    <div class="row">
+      <div class="center-block">
+        <h1>Songs</h1>
+        <input type="search" id="songs-search-field" class="form-control" placeholder="Enter a lyric phrase, artist, or song title"/>
+        <table class="songs-table">
+          <thead>
+            <tr>
+              <th>Name</th>
+              <th>Artist</th>
+              <th>Tempo</th>
+              <th>Key</th>
+            </tr>
+          </thead>
+        </table>
+      </div>
+    </div>
+  </div>
+</div>
 
 <script>
   $(function() {
-    $('.songs-table').DataTable({
+    var table = $('.songs-table').DataTable({
       sAjaxSource: '<%= songs_path(format: :json) %>',
       serverSide: true,
-      processing: true,
       responsive: true,
+      lengthChange: false,
+      dom: '',
       columns: [
         { data: 'name' },
         { data: 'artist' },
         { data: 'tempo' },
         { data: 'key' },
       ],
     });
+
+    $('#songs-search-field').keyup(function() {
+      table.search(this.value).draw();
+    });
+
+    $('.search-page').addClass('active');
```

Comment:
reminder for myself to style the `active` class so that it appears slightly darker than the rest.